[visionOS] Warning in LinearMediaPlayer.swift: "Private module 'LinearMediaKit' is imported publicly from the public module 'libWebKitSwift'"

aestes opened the pull request:
#### f9833d60b73b7476fdcd95a029311b40d15ee901
<pre>
[visionOS] Warning in LinearMediaPlayer.swift: &quot;Private module &apos;LinearMediaKit&apos; is imported publicly from the public module &apos;libWebKitSwift&apos;&quot;
<a href="https://bugs.webkit.org/show_bug.cgi?id=290143">https://bugs.webkit.org/show_bug.cgi?id=290143</a>
<a href="https://rdar.apple.com/147543289">rdar://147543289</a>

Reviewed by Elliott Williams.

Because libWebKitSwift.dylib is installed inside /System/Library/Frameworks/WebKit.framework, the
Xcode build system emits the following warning when Swift sources in WebKitSwift import a private
module, e.g., LinearMediaKit:

    Private module &apos;LinearMediaKit&apos; is imported publicly from the public module &apos;libWebKitSwift&apos;

The build system thinks WebKitSwift is a public module based on its INSTALL_PATH, but in reality
WebKitSwift does not contribute its Swift module to Public or Internal SDKs and should be
considered neither a public nor private module.

Silenced this warning by setting SWIFT_LIBRARY_LEVEL to other.

* Source/WebKit/Configurations/WebKitSwift.xcconfig:

Canonical link: <a href="https://commits.webkit.org/292491@main">https://commits.webkit.org/292491@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/43a464dcc7d87be79a9dd35d02c8ea19ea3d44e2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/96126 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/15740 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/5698 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/101191 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/46635 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/16035 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/24173 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/73279 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/30494 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/99129 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/12011 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/86829 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/53614 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/11761 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/45970 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/81895 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/4687 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/103216 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/23193 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/16889 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/82315 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/23444 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/82848 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/81690 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20520 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/26303 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/3735 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/16549 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/23156 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/28311 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/22815 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/26295 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/24556 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple reviewed (approved):
Seems fine, since WebKit is libWKS’s only client, and it is not part of WebKit’s module interface.

aestes committed:
[visionOS] Warning in LinearMediaPlayer.swift: "Private module 'LinearMediaKit' is imported publicly from the public module 'libWebKitSwift'"
https://bugs.webkit.org/show_bug.cgi?id=290143
rdar://147543289

Reviewed by Elliott Williams.

Because libWebKitSwift.dylib is installed inside /System/Library/Frameworks/WebKit.framework, the
Xcode build system emits the following warning when Swift sources in WebKitSwift import a private
module, e.g., LinearMediaKit:

    Private module 'LinearMediaKit' is imported publicly from the public module 'libWebKitSwift'

The build system thinks WebKitSwift is a public module based on its INSTALL_PATH, but in reality
WebKitSwift does not contribute its Swift module to Public or Internal SDKs and should be
considered neither a public nor private module.

Silenced this warning by setting SWIFT_LIBRARY_LEVEL to other.

* Source/WebKit/Configurations/WebKitSwift.xcconfig:

Canonical link: https://commits.webkit.org/292491@main
